sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -62,6 +62,32 @@ class DataFrameStatSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("randomSplit on reordered partitions") {
+    val n = 600
```

Comment:
also you can just run it twice and make sure the result is deterministic, i.e.
```
val a = df.randomSplit(...).toSeq.map(_.collect())
val b = df.randomSplit(...).toSeq.map(_.collect())
assert(a == b)
```

as long as these are scala collections, I think they will work.